### PR TITLE
Multiple changes for multiple people

### PR DESCRIPTION
**Pull request by ondrae**

### CODE REVIEW

**Comment by ondrae** [comment]
Members will need to change to be allowed to belong to many different changes.

**Comment by ondrae** [comment]
Back button bugs
screen shot 2019-02-20 at 10 59 54 am

**Comment by ondrae** [comment]
@luigi This should be ready for review now. I could use some review on the second commit, trying to address the back button issues. https://github.com/codeforamerica/colorado-benefits/pull/175/commits/c5e23274bacbb8408d07cc61b61e5af93883e0f2

There is a review app at https://colorado-benefits-stagi-pr-175.herokuapp.com. Just can't upload docs. Default admin login works.

File: spec/models/analytics_data_spec.rb [hunk 1]
```diff
@@ -3,45 +3,69 @@
...
+      #   paid_how_often: "monthly",
+      #   paid_yet: "no",
+      #   first_day: DateTime.new(2016, 1, 3),
+      #   member: member2
+      # create :change_navigator, is_self_employed: "no", change: change2
+      # create :change,
+      #   change_type: "job_termination",
+      #   last_paycheck: DateTime.new(2017, 1, 2),
+      #   last_day: DateTime.new(2016, 1, 2),
+      #   member: member3
```
Comment (luigi) [inline spec/models/analytics_data_spec.rb]: Looks like this commented out code was left inadvertently?

File: spec/support/shared_examples/forms_controller_change_in_hours_behavior.rb [hunk 2]
```diff
@@ -4,19 +4,32 @@
...
+    before do
+      member = create :member, report: report
+      change = create :change,
+        change_type: change_type,
+        member: member
+      create :navigator,
+        current_member: member,
+        current_change: change,
+        report: report
+    end
```
Comment (luigi) [inline spec/support/shared_examples/forms_controller_change_in_hours_behavior.rb]: Is this still needed now that the `:filled` trait is on the `report` factory?

File: app/models/navigator.rb [hunk 3]
```diff
@@ -1,9 +1,13 @@
 class Navigator < ActiveRecord::Base
   belongs_to :report
+  belongs_to :current_member, class_name: "Member", foreign_key: "current_member_id", optional: true
+  belongs_to :current_change, class_name: "Change", foreign_key: "current_change_id", optional: true
 
   enum selected_county_location: { unfilled: 0, arapahoe: 1, not_arapahoe: 2, not_sure: 3 },
        _prefix: :selected_county_location
   enum submitting_for: { unfilled: 0, self: 1, other_member: 2 }, _prefix: :submitting_for
+  enum temp_change_type: { unfilled: 0, job_termination: 1, new_job: 2, change_in_hours: 3 },
+       _prefix: :temp_change_type
```
Comment (luigi) [inline app/models/navigator.rb]: I wonder if calling it `selected_change_type` is more descriptive?

Just something to keep in mind. No need to change in this PR.

File: app/controllers/change_type_controller.rb [hunk 4]
```diff
@@ -1 +1,6 @@
-class ChangeTypeController < FormsController; end
+class ChangeTypeController < FormsController
+  def update_session
+    # clear out current member and change
+    current_report.navigator.update(current_member_id: nil, current_change_id: nil)
```
Comment (luigi) [inline app/controllers/change_type_controller.rb]: If we use this pattern a third time, we should abstract it out to a method like `current_report.navigator.clear_currents`.

Comment (ondrae) [reply app/controllers/change_type_controller.rb]: I actually got rid of one of the other places where we cleared it out. It was the cause of some back button bugs. So this is the only place we clear out the current_member and current_change now.

File: spec/support/shared_examples/forms_controller_change_in_hours_behavior.rb (hunk 2 above)
Comment (ondrae) [reply spec/support/shared_examples/forms_controller_change_in_hours_behavior.rb]: Cleaned up all these behavior specs. Thanks.

File: app/models/navigator.rb (hunk 3 above)
Comment (ondrae) [reply app/models/navigator.rb]: I made the change.

**Pull request merged**